java/ql/src/semmle/code/java/security/XPath.qll

```diff
@@ -0,0 +1,65 @@
+/** Provides classes to reason about XPath vulnerabilities. */
+
+import java
+import semmle.code.java.dataflow.FlowSources
+import semmle.code.java.dataflow.TaintTracking
+
+/**
+ * An abstract type representing a call to interpret XPath expressions.
+ */
+class XPathSink extends MethodAccess {
```

Comment:
It seems superfluous to have both this class and `XPathInjectionSink`, so let's get rid of this one. Also, let's make as many definitions as possible `private` within this file.

Reply:
Got it, thanks! Regarding visibility, I changed almost everything to `private` - I guess we can always change any of the componets back to public if we need them elsewhere. See 2054791